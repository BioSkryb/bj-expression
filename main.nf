nextflow.enable.dsl=2

include { printHeader; helpMessage } from './help' params ( params )
include { RNASEQ_WF } from './workflow/rnaseq.nf' params ( params )


if ( params.help ) {
    helpMessage()
    exit 0
}



params.salmon_index               = WorkflowscRNASeq.getGenomeAttribute( params, 'salmon_index' )
params.tx2gene                    = WorkflowscRNASeq.getGenomeAttribute( params, 'tx2gene' )
params.star_index                 = WorkflowscRNASeq.getGenomeAttribute( params, 'star_index' )
params.gtf_file                   = WorkflowscRNASeq.getGenomeAttribute( params, 'gtf_file' )
params.genebody_ref               = params.genomes [ params.genome ] [ 'genebody_ref']

workflow {
    
    printHeader()
    if ( params.reads ) {
                
        ch_reads = Channel.fromFilePairs( params.reads , size: -1 , checkExists: true )
        ch_reads.ifEmpty{ exit 1, "ERROR: cannot find any fastq files matching the pattern: ${params.reads}\nMake sure that the input file exists!" }

    } else if ( params.input_csv  ) {
                
        ch_reads = Channel.fromPath( params.input_csv  ).splitCsv( header:true )
                        .map { row -> [ row.biosampleName, [ row.read1, row.read2 ] ] }
        ch_reads.ifEmpty{ exit 1, "ERROR: Input csv file is empty." }
    }


                        
     ch_multiqc_config = Channel.fromPath(params.multiqc_config, checkIfExists: true)
     ch_reference_celltype = Channel.fromPath(params.celltype_ref)    

 
       
    RNASEQ_WF( 
                params.publish_dir,
                params.enable_publish,
                params.disable_publish,
                ch_reads, 
                params.input_csv,
                params.adapter_sequence,
                params.adapter_sequence_r2,
                params.salmon_index,
                params.star_index, 
                params.gtf_file, 
                params.tx2gene,
                ch_reference_celltype,
                ch_multiqc_config,
                params.project,
                params.genebody_ref

             )
}

// OnComplete
workflow.onComplete{
    println( "\nPipeline completed successfully.\n\n" )
}

// OnError
workflow.onError{
    println( "\nPipeline failed.\n\n" )
}