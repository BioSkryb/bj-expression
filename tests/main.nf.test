nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"

    test("scrna-seq test") {

        when {
            params {
                // define parameters here. Example: 
                publish_dir = "${outputDir}/results"
                timestamp = "test"
                architecture = "x86"
            }
        }

        then {
            assertAll(
                // Check if the workflow was successful
                { assert workflow.success },

                // Verify existence of the multiqc report HTML file
                {assert new File("${outputDir}/results_test/multiqc/multiqc_report.html").exists()},

                // Check for a match in the pipeline_metrics_summary csv file
                {assert snapshot (path("${outputDir}/results_test/secondary_analyses/secondary_metrics/qualimap_stats_mqc.csv")).match("pipeline_metrics_summary")},

                // Check for a match in the pipeline_metrics_summary_percents csv file
                {assert snapshot (path("${outputDir}/results_test/secondary_analyses/secondary_metrics/pipeline_metrics_summary_percents.csv")).match("pipeline_metrics_summary_percents")},

                // Check for a match in the df_dynamicrange_expression tsv file
                {assert snapshot (path("${outputDir}/results_test/secondary_analyses/secondary_metrics/df_dynamicrange_expression.tsv")).match("df_dynamicrange_expression")},

                // Verify existence of the df_gene_counts_salmon file
                {assert new File("${outputDir}/results_test/secondary_analyses/quantification_salmon/df_gene_counts_salmon.tsv").exists()},

                // Verify existence of the df_gene_counts_starhtseq file
                {assert new File("${outputDir}/results_test/secondary_analyses/quantification_htseq/df_gene_counts_starhtseq.tsv").exists()},

                // Verify existence of the bam file
                {assert new File("${outputDir}/results_test/secondary_analyses/alignment_htseq/Expression-test1.bam.bai").exists()}

            )
        }

    }

}
