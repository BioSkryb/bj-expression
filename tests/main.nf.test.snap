{
    "pipeline_metrics_summary": {
        "content": [
            "qualimap_stats_mqc.csv:md5,f9f9bcb755b3646c8382d379aac86c4a"
        ],
        "timestamp": "2024-05-14T14:36:14+0000"
    },
    "pipeline_metrics_summary_percents": {
        "content": [
            "pipeline_metrics_summary_percents.csv:md5,0cef2eeeb9d50a65000b7ab8109f7e13"
        ],
        "timestamp": "2024-05-14T14:36:14+0000"
    },
    "df_dynamicrange_expression": {
        "content": [
            "df_dynamicrange_expression.tsv:md5,108e27000a74a17170f9c5dfc7dedc82"
        ],
        "timestamp": "2024-05-14T14:36:14+0000"
    }
}